core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -210,6 +210,16 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
     }
   }
 
+  def truncateCheckpoint(dataDir: File, topicAndPartition: TopicAndPartition, offset: Long) {
+    inLock(lock) {
+      val checkpoint = checkpoints(dataDir)
+      val existing = checkpoint.read()
+
+      if (existing.getOrElse(topicAndPartition, 0L) > offset)
```

Comment:
I see. I will update the patch to only read checkpoint file it the topic is compacted.